Source/Diagnostics/SliceDiagnostic.cpp

```diff
@@ -0,0 +1,490 @@
+                                         scomp, ncomp, cr_ratio);
+           }
+           if( SliceType == WarpX::Ex_nodal_flag  ) {
+              amrex::amrex_avgdown_edges(Dst_bx, Dst_fabox, Src_fabox, dcomp, 
+                                         scomp, ncomp, cr_ratio, 0);
+           }
+           if( SliceType == WarpX::Ey_nodal_flag) {
+              amrex::amrex_avgdown_edges(Dst_bx, Dst_fabox, Src_fabox, dcomp, 
+                                         scomp, ncomp, cr_ratio, 1);
+           }
+           if( SliceType == WarpX::Ez_nodal_flag ) {
+              amrex::amrex_avgdown_edges(Dst_bx, Dst_fabox, Src_fabox, dcomp,
+                                         scomp, ncomp, cr_ratio, 2);
+           }
+           if( SliceType == WarpX::Bx_nodal_flag) {
+              amrex::amrex_avgdown_faces(Dst_bx, Dst_fabox, Src_fabox, dcomp, 
+                                         scomp, ncomp, cr_ratio, 0);
+           }
+           if( SliceType == WarpX::By_nodal_flag ) {
+              amrex::amrex_avgdown_faces(Dst_bx, Dst_fabox, Src_fabox, dcomp, 
+                                         scomp, ncomp, cr_ratio, 1);
+           }
+           if( SliceType == WarpX::Bz_nodal_flag ) {
+              amrex::amrex_avgdown_faces(Dst_bx, Dst_fabox, Src_fabox, dcomp, 
+                                         scomp, ncomp, cr_ratio, 2);
+           }
+
+           if ( mfi_dst.isValid() ) {           
+              ++mfi_dst;
+           }
+
+       }
+       return cs_mf;
+
+    }
+    
+    return smf;
+}
+
+
+/* \brief
+ *  This function modifies the slice input parameters under certain conditions.  
+ *  The coarsening ratio, slice_cr_ratio is modified if the input is not an exponent of 2. 
+ *  for example, if the coarsening ratio is 3, 5 or 6, which is not an exponent of 2, 
+ *  then the value of coarsening ratio is modified to the nearest exponent of 2. 
+ *  The default value for coarsening ratio is 1. 
+ *  \param slice_realbox a Real box for defining the slice dimension. 
+ *  slice_realbox.lo and slice_realbox.hi are set equal to the simulation domain lo and hi 
+ *  if for the user-input for the slice lo and hi coordinates are outside the domain. 
+ *  If the slice_realbox.lo and slice_realbox.hi coordinates do not align with the data 
+ *  points and the number of cells in that dimension is greater than 1, and if the extent of
+ *  the slice in that dimension is not coarsenable, then the value lo and hi coordinates are
+ *  shifted to the nearest coarsenable point to include some extra data points in the slice.
+ *  If slice_realbox.lo==slice_realbox.hi, then that dimension has only one cell and no
+ *  modifications are made to the value. If the lo and hi do not align with a data point, 
+ *  then it is flagged for interpolation.  
+ *  \param slice_lo and slice_hi are the index values of the slice
+ *  The slice shares the same index space as that of the simulation domain. 
+ */
```

Comment:
Thanks fo the docstring!
Again, it would be nice if you could match the `\param` to the function inputs. 

Note that this is not a priority: we may add the capability to have several `slice` objects in the future. For this, one option would be to create a new class, e.g., `SlicePlot`, of which these functions would be member functions. In this case, most of the parameters passed here may end up being member data of the class, so there we would not need these `\param` in the docstring.

Reply:
Okay. For now I will define the \param and remove the definitions after we modify for multiple slice objects.